src/entry.ts

```diff
@@ -273,9 +275,22 @@ class Entry {
         ? meta.traceSampled
         : undefined;
     // Format log payload.
-    entry.message =
+    let data =
       meta.textPayload || meta.jsonPayload || meta.protoPayload || undefined;
-    entry.message = this.data || entry.message;
+    if (useMessageField) {
+      entry.message = this.data || data;
+    } else {
+      data = this.data || data;
```

Comment:
nit: this block could probably use some comments

Reply:
Added